deploy/README.md

```diff
@@ -51,3 +51,21 @@ up volume file permissions for the circleci UID (specified in our Dockerfile).
 **Note**: Local changes to deploy.rb or devstart.rb are **not** currently picked
 up by the deployment process; it uses the tools code from the commit version
 specified for deployment.
+
+# Manual testing
+
+Note: See above for an explanation of why this process is special right now.
+This could be improved in the future by supporting builds via dirtly clients.
+
+Use the --dry-run flag to test changes. To ensure your local changes are picked
+up during the deploy process, follow this process:
+
+1. Commit your changes to the deploy scripts
+1. Push your branch to GitHub, e.g. 'ch/deploy-fix'
+1. Use --dry-run and --git-version to build using your branch:
```

Comment:
So the point of dry-run is just to check to make sure your script is issuing the commands you expected?

I can see how that could be useful for quick iteration on script changes, but you probably won't have confidence that everything works as intended until you actually run it.

Reply:
Good point, I've added instructions here as well for if you want to do a full test deployment (--no-promote)